### PR TITLE
Add a guide how to handle complex expressions.

### PR DESCRIPTION
**Pull request by Xazax-hun**

### CODE REVIEW

File: docs/false_positives.md [hunk 1]
```diff
@@ -176,6 +176,39 @@ an important invariant of the program explicit. Moreover, it will make the
 false positive finding disappear. This will also provide the users of the code
 with an explicit check in the debug build which can help find bugs.
 
+In some cases the analyzer cannot reason about the values of expressions
+due to some limitations of the constraint solver. In the code example 
+bellow the analyzer cannot record the constraint about a complex expression.
```
Comment (rnkovacs) [inline docs/false_positives.md]: Nit: be**l**ow

File: docs/false_positives.md [hunk 2]
```diff
@@ -176,6 +176,39 @@ an important invariant of the program explicit. Moreover, it will make the
 false positive finding disappear. This will also provide the users of the code
 with an explicit check in the debug build which can help find bugs.
 
+In some cases the analyzer cannot reason about the values of expressions
```
Comment (whisperity) [inline docs/false_positives.md]: Maybe "*possible* values" instead?

**Comment by whisperity** [comment]
OFF: Are these practices available elsewhere too? Is this collection general and broad enough? Does the examples here constitute original research?

If so, perhaps it'd be worth a shot to put (at least parts of) this guide into ClangSA's documentation too in the LLVM repo?

**Comment by Xazax-hun** [comment]
@whisperity There are some relevant documentation at the Clang Static Analyzer's page. But we added new information that is not available there. I do plan to send this over the CSA maintainers so they can take a look at if they would like to include some of this info on the official page.

**Pull request merged**